### PR TITLE
#159965482 Rate articles.

### PR DESCRIPTION
**Pull request by Mutuba**
## What does this PR do?
```
Creates a functionality to enable users to rate articles authored by others.
```

## Description of Task to be completed?
```
Users should be able to rate articles authored by others.
An article can be rated on the of 1 to 5.
Depending on numbers of raters per article, an average should be calculated and returned.

```

## How should this be manually tested?

```
Clone the repo, cd into ah-leagueOfLegends. Create two accounts. Create articles using HTTP POST on:
http://127.0.0.1:8000/api/articles. Using the other account, rate the article using HTTP POST on: 
http://127.0.0.1:8000/api/articles/your-article-slug/rate/ with the below sample data:
{
  "rate": {
    "rating": 5
  }
}

```
## Any background context you want to provide?

```
AuthorsHaven did not have a rating functionality for articles

```

## What are the relevant pivotal tracker stories?

[#159965482](https://www.pivotaltracker.com/n/projects/2192365)

## Screenshots if available to support PR.

screen shot 2018-09-19 at 12 58 35
screen shot 2018-09-19 at 13 00 14

## Questions:

```
What do you think of the implementation?
Is there a way to improve on the current functionality?

```

### CODE REVIEW

File: authors/apps/articles/test_articles_ratings.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+        article= {
+                    "article": 
+                      {
+                      "title": "How to feed your dragon",
+                      "description": "Wanna know how?",
+                      "body": "You don't believe?",
+                        }
+                  }
+
```
Comment (Quantum-35) [inline authors/apps/articles/test_articles_ratings.py]: I'll suggest you take that article to the base file.

File: authors/apps/articles/test_articles_ratings.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+            self.SIGN_UP_URL,
+            self.user_cred,
+            format='json')
+        response = self.client.post(
+            reverse('authentication:user_login'),
+            self.user_cred,
+            format='json')
+        token = response.data['token']
+
+        article = {
```
Comment (Quantum-35) [inline authors/apps/articles/test_articles_ratings.py]: Take this article declaration to the base file so that it can be reused or edited with ease.

**Review by Quantum-35 — APPROVED** [review]

File: authors/apps/articles/models.py [hunk 3]
```diff
@@ -68,6 +69,12 @@ def save(self, *args, **kwargs):
         self.slug = self._get_unique_slug()
         super(Article, self).save(*args, **kwargs)
 
+    def updaterate(self, rating):
+        '''
```
Comment (Teddykavooh) [inline authors/apps/articles/models.py]: Do away with the empty doc strings.

**Review by Teddykavooh — APPROVED** [review]
Good work.

**Pull request merged**